The new `defaultInput` breaks multiple generators

The new `defaultInput` structure with multiple levels of inheritance with `allOf` breaks in multiple generators.

The following generators are confirmed to break at the generation stage:
- `spring` (server)
- `python` (client)
- `java` (client)
- `csharp` (client)

The following generators break when compiling/building generated code:
- `aspnetcore` (server)
- `kotlin-spring` (server)
- `spring` (server), since there's nothing to build
- `python` (client), since there's nothing to build
- `kotlin` (client)
- `java` (client), since there's nothing to build
- `csharp` (client), since there's nothing to build
- `typescript-axios` (client)

If not clear by looking at `defaultInput`, this problem extends to anything with more than 1 level of `allOf` inheritance. E.g. errors from the `typescript-axios` generator include:
```bash
api.ts(133,18): error TS2430: Interface 'AncillaryInput' incorrectly extends interface 'LegInput'.
  Types of property ''type'' are incompatible.
    Type '"ancillary"' is not assignable to type '"leg"'.
api.ts(183,18): error TS2430: Interface 'AssetInput' incorrectly extends interface 'LegInput'.
  Types of property ''type'' are incompatible.
    Type '"asset"' is not assignable to type '"leg"'.
api.ts(534,18): error TS2430: Interface 'ExtendExpiryTimeInput' incorrectly extends interface 'PackageInput'.
  Types of property ''type'' are incompatible.
    Type '"extend_expiry_time"' is not assignable to type '"package"'.
api.ts(811,18): error TS2430: Interface 'LegInput' incorrectly extends interface 'PackageInput'.
  Types of property ''type'' are incompatible.
    Type '"leg"' is not assignable to type '"package"'.
api.ts(1397,18): error TS2430: Interface 'PurchasePackageInput' incorrectly extends interface 'SearchOfferInput'.
  Types of property ''type'' are incompatible.
    Type '"purchase_package"' is not assignable to type '"search_offer"'.
api.ts(1428,18): error TS2430: Interface 'RedressInput' incorrectly extends interface 'PackageInput'.
  Types of property ''type'' are incompatible.
    Type '"redress"' is not assignable to type '"package"'.
```

--- a/OMSA.yaml
+++ b/OMSA.yaml
@@ -1504,24 +1504,12 @@ components:
           type: string
           enum: [ search_offer ]
         specification:
-          type: object
-          properties:
-            from:
-              $ref: "#/components/schemas/externalReference"
-            via:
-              type: array
-              minItems: 0
-              maxItems: 3
-              items:
-                $ref: "#/components/schemas/externalReference"
-            to:
-              $ref: "#/components/schemas/externalReference"
-            startTime:
-              $ref: "#/components/schemas/dateTime"
-            endTime:
-              $ref: "#/components/schemas/dateTime"
-            placeDefinitions:
-              $ref: "#/components/schemas/placeDefinitions"
+          allOf:
+          - $ref: "#/components/schemas/requestedTravelSpecification"
+          - type: object
+            properties:
+              placeDefinitions:
+                $ref: "#/components/schemas/placeDefinitions"
         pattern:
           $ref: "#/components/schemas/tripPattern"
         packageToExchange:
@@ -1694,24 +1682,12 @@ components:
           type: string
           enum: [ purchase_package ]
         specification:
-          type: object
-          properties:
-            from:
-              $ref: "#/components/schemas/externalReference"
-            via:
-              type: array
-              minItems: 0
-              maxItems: 3
-              items:
-                $ref: "#/components/schemas/externalReference"
-            to:
-              $ref: "#/components/schemas/externalReference"
-            startTime:
-              $ref: "#/components/schemas/dateTime"
-            endTime:
-              $ref: "#/components/schemas/dateTime"
-            placeDefinitions:
-              $ref: "#/components/schemas/placeDefinitions"
+          allOf:
+          - $ref: "#/components/schemas/requestedTravelSpecification"
+          - type: object
+            properties:
+              placeDefinitions:
+                $ref: "#/components/schemas/placeDefinitions"
         pattern:
           $ref: "#/components/schemas/tripPattern"
         packageToExchange:
@@ -1749,24 +1725,12 @@ components:
           $ref: "#/components/schemas/customer"
           description: The customer making the purchase. This can be different from the travellers.
         specification:
-          type: object
-          properties:
-            from:
-              $ref: "#/components/schemas/externalReference"
-            via:
-              type: array
-              minItems: 0
-              maxItems: 3
-              items:
-                $ref: "#/components/schemas/externalReference"
-            to:
-              $ref: "#/components/schemas/externalReference"
-            startTime:
-              $ref: "#/components/schemas/dateTime"
-            endTime:
-              $ref: "#/components/schemas/dateTime"
-            placeDefinitions:
-              $ref: "#/components/schemas/placeDefinitions"
+          allOf:
+          - $ref: "#/components/schemas/requestedTravelSpecification"
+          - type: object
+            properties:
+              placeDefinitions:
+                $ref: "#/components/schemas/placeDefinitions"
         pattern:
           $ref: "#/components/schemas/tripPattern"
         packageToExchange:

--- a/OMSA.yaml
+++ b/OMSA.yaml
@@ -339,7 +339,7 @@ paths:
               type: object
               properties:
                 inputs:
-                  $ref: "#/components/schemas/packageInput"
+                  $ref: "#/components/schemas/purchaseOffersInput"
                 subscriber:
                   $ref: "#/components/schemas/subscriber"
       responses:
@@ -381,7 +381,7 @@ paths:
               type: object
               properties:
                 inputs:
-                  $ref: "#/components/schemas/packageInput"
+                  $ref: "#/components/schemas/purchasePackageInput"
                 subscriber:
                   $ref: "#/components/schemas/subscriber"
       responses:
@@ -507,7 +507,7 @@ paths:
               type: object
               properties:
                 inputs:
-                  $ref: "#/components/schemas/packageInput"
+                  $ref: "#/components/schemas/extendExpiryTimeInput"
                 subscriber:
                   $ref: "#/components/schemas/subscriber"
       responses:
@@ -1581,7 +1581,7 @@ components:
           type:
             type: string
             enum: [ package ]
-
+    
     legInput:
       allOf:
       - $ref: "#/components/schemas/packageInput"

--- a/OMSA.yaml
+++ b/OMSA.yaml
@@ -1581,7 +1581,7 @@ components:
           type:
             type: string
             enum: [ package ]
-    
+
     legInput:
       allOf:
       - $ref: "#/components/schemas/packageInput"

--- a/OMSA.yaml
+++ b/OMSA.yaml
@@ -1284,24 +1284,6 @@ components:
           type: string
           format: uri
 
-    defaultInput:
-      type: object
-      discriminator:
-        propertyName: type
-        mapping:
-          SEARCH_OFFER: "#/components/schemas/searchOfferInput"
-          PURCHASE: "#/components/schemas/packageInput"
-          COMMIT: "#/components/schemas/packageInput"
-          RELEASE: "#/components/schemas/packageInput"
-          PURCHASE_PACKAGE: "#/components/schemas/purchasePackageInput"
-          PURCHASE_OFFERS: "#/components/schemas/purchaseOffersInput"
-          EXTEND_EXPIRY_TIME: "#/components/schemas/extendExpiryTimeInput"
-      required:
-      - type
-      properties:
-        type:
-          type: string
-
     landingPage:
       type: object
       required:
@@ -1512,116 +1494,81 @@ components:
 
     # Inputs
     searchOfferInput:
-      allOf:
-      - $ref: "#/components/schemas/defaultInput"
-      - type: object
-        required:
-        - travellers
-        - type
-        description: A package planning request, resulting in package options
-        properties:
-          type:
-            type: string
-            enum: [ search_offer ]
-          specification:
-            allOf:
-            - $ref: "#/components/schemas/requestedTravelSpecification"
-            - type: object
-              properties:
-                placeDefinitions:
-                  $ref: "#/components/schemas/placeDefinitions"
-          pattern:
-            $ref: "#/components/schemas/tripPattern"
-          packageToExchange:
-            $ref: "#/components/schemas/packageReference"
-          travellers:
-            type: array
-            maxItems: 10
-            minItems: 1
-            items:
-              $ref: "#/components/schemas/individualTraveller"
-          profiles:
-            type: array
-            items:
-              $ref: "#/components/schemas/userProfile"
-          travellingAssets:
-            type: array
-            items:
-              $ref: "#/components/schemas/travellingAsset"
+      type: object
+      required:
+      - travellers
+      - type
+      description: A package planning request, resulting in package options
+      properties:
+        type:
+          type: string
+          enum: [ search_offer ]
+        specification:
+          type: object
+          properties:
+            from:
+              $ref: "#/components/schemas/externalReference"
+            via:
+              type: array
+              minItems: 0
+              maxItems: 3
+              items:
+                $ref: "#/components/schemas/externalReference"
+            to:
+              $ref: "#/components/schemas/externalReference"
+            startTime:
+              $ref: "#/components/schemas/dateTime"
+            endTime:
+              $ref: "#/components/schemas/dateTime"
+            placeDefinitions:
+              $ref: "#/components/schemas/placeDefinitions"
+        pattern:
+          $ref: "#/components/schemas/tripPattern"
+        packageToExchange:
+          $ref: "#/components/schemas/packageReference"
+        travellers:
+          type: array
+          maxItems: 10
+          minItems: 1
+          items:
+            $ref: "#/components/schemas/individualTraveller"
+        profiles:
+          type: array
+          items:
+            $ref: "#/components/schemas/userProfile"
+        travellingAssets:
+          type: array
+          items:
+            $ref: "#/components/schemas/travellingAsset"
 
     selectOffersInput:
-      allOf:
-      - $ref: "#/components/schemas/defaultInput"
-      - type: object
-        required:
-        - type
-        - offerIds
-        properties:
-          offerIds:
-            type: array
-            items:
-              $ref: "#/components/schemas/offerReference"
-          type:
-            type: string
-            enum: [ select_offers ]
-          customer:
-            $ref: "#/components/schemas/customer"
-            description: The customer making the purchase. This can be different from the travellers.
+      type: object
+      required:
+      - type
+      - offerIds
+      properties:
+        type:
+          type: string
+          enum: [ select_offers ]
+        offerIds:
+          type: array
+          items:
+            $ref: "#/components/schemas/offerReference"
+        customer:
+          $ref: "#/components/schemas/customer"
+          description: The customer making the purchase. This can be different from the travellers.
 
     packageInput:
-      allOf:
-      - $ref: "#/components/schemas/defaultInput"
-      - type: object
-        required:
-        - type
-        - packageId
-        properties:
-          packageId:
-            $ref: "#/components/schemas/packageReference"
-          type:
-            type: string
-            enum: [ package ]
-
-    legInput:
-      allOf:
-      - $ref: "#/components/schemas/packageInput"
-      - type: object
-        required:
-        - legId
-        - type
-        properties:
-          offerId:
-            $ref: "#/components/schemas/offerReference"
-          legId:
-            $ref: "#/components/schemas/legReference"
-          type:
-            type: string
-            enum: [ leg ]
-          location:
-            $ref: "#/components/schemas/placeReference"
-         #evidence:
-         #   type: array
-         #   minItems: 1
-         #   items:
-         #     $ref: "#/components/schemas/link"
-
-    refundInput:
-      allOf:
-      - $ref: "#/components/schemas/legInput"
-      - type: object
-        required:
-          - type
-        properties:
-          type:
-            type: string
-            enum: [ refund-option ]
-          reason:
-            $ref: "#/components/schemas/normalString"
-          refundType:
-            type: string
-            enum: [ package_refund, remove_traveller, remove_ancillary ]
-          ancillary:
-            $ref: "#/components/schemas/ancillaryReference"
+      type: object
+      required:
+      - type
+      - packageId
+      properties:
+        type:
+          type: string
+          enum: [ package ]
+        packageId:
+          $ref: "#/components/schemas/packageReference"
 
     refundOptionInput:
       type: object
@@ -1655,107 +1602,224 @@ components:
             - $ref: "#/components/schemas/travellingAsset"
 
     travelSpecificationInput:
-      allOf:
-      - $ref: "#/components/schemas/packageInput"
-      - $ref: "#/components/schemas/travelSpecification"
-      - type: object
-        required:
-          - type
-          - packageId
-        properties:
-          legId:
-            $ref: "#/components/schemas/legReference"
-          type:
-            type: string
-            enum: [ change_times ]
+      type: object
+      required:
+        - type
+        - packageId
+      properties:
+        type:
+          type: string
+          enum: [ change_times ]
+        packageId:
+          $ref: "#/components/schemas/packageReference"
+        legId:
+          $ref: "#/components/schemas/legReference"
+        from:
+          $ref: "#/components/schemas/placeReference"
+          description: use an ID contained in the **placeDefinitions** field, or from an external source, when using coordinates, please use the prefix 'gps:'
+        via:
+          type: array
+          description: use an ID contained in the **placeDefinitions** field, or from an external source, when using coordinates, please use the prefix 'gps:'
+          minItems: 0
+          maxItems: 3
+          items:
+            $ref: "#/components/schemas/placeReference"
+        to:
+          $ref: "#/components/schemas/placeReference"
+          description: use an ID contained in the **placeDefinitions** field, or from an external source, when using coordinates, please use the prefix 'gps:'
+        startTime:
+          $ref: "#/components/schemas/dateTime"
+          description: The intended departure time.
+            If left out and no endTime is set, the current time should be assumed.
+            If only the arrival time is specified, this is an implicit request for a guaranteed arrival at that time.
+        endTime:
+          $ref: "#/components/schemas/dateTime"
+          description: The intended arrival time, at the `to place`. When the **startTime** is not set, and **endTime** is set, it is an implicit request for an arrival time guarantee.
 
     assetInput:
-      allOf:
-      - $ref: "#/components/schemas/legInput"
-      - type: object
-        required:
-        - type
-        - assetId
-        properties:
-          type:
-            type: string
-            enum: [ asset ]
-          assetId:
-            $ref: "#/components/schemas/assetReference"
-          replaceAssetId:
-            $ref: "#/components/schemas/assetReference"
+      type: object
+      required:
+      - type
+      - assetId
+      - packageId
+      - legId
+      properties:
+        type:
+          type: string
+          enum: [ asset ]
+        assetId:
+          $ref: "#/components/schemas/assetReference"
+        replaceAssetId:
+          $ref: "#/components/schemas/assetReference"
+        packageId:
+          $ref: "#/components/schemas/packageReference"
+        offerId:
+          $ref: "#/components/schemas/offerReference"
+        legId:
+          $ref: "#/components/schemas/legReference"
+        location:
+          $ref: "#/components/schemas/placeReference"
 
     ancillaryInput:
-      allOf:
-      - $ref: "#/components/schemas/legInput"
-      - type: object
-        required:
-        - type
-        - ancillaryId
-        properties:
-          type:
-            type: string
-            enum: [ ancillary ]
-          ancillaryId:
-            $ref: "#/components/schemas/ancillaryReference"
-          replaceAncillaryId:
-            $ref: "#/components/schemas/ancillaryReference"
+      type: object
+      required:
+      - type
+      - ancillaryId
+      - packageId
+      - legId
+      properties:
+        type:
+          type: string
+          enum: [ ancillary ]
+        ancillaryId:
+          $ref: "#/components/schemas/ancillaryReference"
+        replaceAncillaryId:
+          $ref: "#/components/schemas/ancillaryReference"
+        packageId:
+          $ref: "#/components/schemas/packageReference"
+        offerId:
+          $ref: "#/components/schemas/offerReference"
+        legId:
+          $ref: "#/components/schemas/legReference"
+        location:
+          $ref: "#/components/schemas/placeReference"
 
     purchasePackageInput:
-      allOf:
-        - $ref: "#/components/schemas/searchOfferInput"
-        - type: object
-          required:
-          - type
+      type: object
+      required:
+      - travellers
+      - type
+      properties:
+        type:
+          type: string
+          enum: [ purchase_package ]
+        specification:
+          type: object
           properties:
-            type:
-              type: string
-              enum: [ purchase_package ]
+            from:
+              $ref: "#/components/schemas/externalReference"
+            via:
+              type: array
+              minItems: 0
+              maxItems: 3
+              items:
+                $ref: "#/components/schemas/externalReference"
+            to:
+              $ref: "#/components/schemas/externalReference"
+            startTime:
+              $ref: "#/components/schemas/dateTime"
+            endTime:
+              $ref: "#/components/schemas/dateTime"
+            placeDefinitions:
+              $ref: "#/components/schemas/placeDefinitions"
+        pattern:
+          $ref: "#/components/schemas/tripPattern"
+        packageToExchange:
+          $ref: "#/components/schemas/packageReference"
+        travellers:
+          type: array
+          maxItems: 10
+          minItems: 1
+          items:
+            $ref: "#/components/schemas/individualTraveller"
+        profiles:
+          type: array
+          items:
+            $ref: "#/components/schemas/userProfile"
+        travellingAssets:
+          type: array
+          items:
+            $ref: "#/components/schemas/travellingAsset"
 
     purchaseOffersInput:
-      allOf:
-        - $ref: "#/components/schemas/selectOffersInput"
-        - $ref: "#/components/schemas/purchasePackageInput"
-        - type: object
-          required:
-          - type
+      type: object
+      required:
+      - type
+      - travellers
+      - offerIds
+      properties:
+        type:
+          type: string
+          enum: [ purchase_offers ]
+        offerIds:
+          type: array
+          items:
+            $ref: "#/components/schemas/offerReference"
+        customer:
+          $ref: "#/components/schemas/customer"
+          description: The customer making the purchase. This can be different from the travellers.
+        specification:
+          type: object
           properties:
-            type:
-              type: string
-              enum: [ purchase_offers ]
+            from:
+              $ref: "#/components/schemas/externalReference"
+            via:
+              type: array
+              minItems: 0
+              maxItems: 3
+              items:
+                $ref: "#/components/schemas/externalReference"
+            to:
+              $ref: "#/components/schemas/externalReference"
+            startTime:
+              $ref: "#/components/schemas/dateTime"
+            endTime:
+              $ref: "#/components/schemas/dateTime"
+            placeDefinitions:
+              $ref: "#/components/schemas/placeDefinitions"
+        pattern:
+          $ref: "#/components/schemas/tripPattern"
+        packageToExchange:
+          $ref: "#/components/schemas/packageReference"
+        travellers:
+          type: array
+          maxItems: 10
+          minItems: 1
+          items:
+            $ref: "#/components/schemas/individualTraveller"
+        profiles:
+          type: array
+          items:
+            $ref: "#/components/schemas/userProfile"
+        travellingAssets:
+          type: array
+          items:
+            $ref: "#/components/schemas/travellingAsset"
 
     redressInput:
-      allOf:
-      - $ref: "#/components/schemas/packageInput"
-      - type: object
-        required:
-        - type
-        - redressOptionId
-        properties:
-          type:
-            type: string
-            enum: [ redress ]
-          redressOptionId:
-            $ref: "#/components/schemas/uuid"
+      type: object
+      required:
+      - type
+      - packageId
+      - redressOptionId
+      properties:
+        type:
+          type: string
+          enum: [ redress ]
+        packageId:
+          $ref: "#/components/schemas/packageReference"
+        redressOptionId:
+          $ref: "#/components/schemas/uuid"
 
     extendExpiryTimeInput:
-      allOf:
-      - $ref: "#/components/schemas/packageInput"
-      - type: object
-        required:
-        - type
-        properties:
-          type:
-            type: string
-            enum: [ extend_expiry_time ]
-          extensionReason:
-            type: string
-            description:
-              in case operation is EXTEND_EXPIRY_TIME, the reason for extension must be supplied here.<br>
-              _PURCHASE_PENDING_ - The internal purchase process on the MP side is not yet finished<br>
-              _PAYMENT_PENDING_ - The customer is in the payment process<br>
-              _OTHER_ - unspecified
-            enum: [ purchase_pending, payment_pending, other ]
+      type: object
+      required:
+      - type
+      - packageId
+      properties:
+        type:
+          type: string
+          enum: [ extend_expiry_time ]
+        packageId:
+          $ref: "#/components/schemas/packageReference"
+        extensionReason:
+          type: string
+          description:
+            in case operation is EXTEND_EXPIRY_TIME, the reason for extension must be supplied here.<br>
+            _PURCHASE_PENDING_ - The internal purchase process on the MP side is not yet finished<br>
+            _PAYMENT_PENDING_ - The customer is in the payment process<br>
+            _OTHER_ - unspecified
+          enum: [ purchase_pending, payment_pending, other ]
 
     # GeoJSON specs, in contradiction to the convention starting with a capital. Reason: compatibility with GeoJSON specification (types are named this way)
     basePoint:

